server.js

```diff
@@ -8,13 +8,52 @@ app.use(cors());
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
-
 app.get("/", function (request, response) {
-  response.send("Hotel booking server.  Ask for /bookings, etc.");
+  response.send({ bookings });
 });
+//Read all bookings
 
+app.get("/booking", function (request, response) {
+  response.send({ bookings });
+});
+//Create a new booking
+app.post("/booking", (req, resp) => {
+  const newBooking = req.body;
+  console.log(newBooking);
+  if (
+    newBooking.id &&
+    newBooking.title &&
+    newBooking.firstName &&
+    newBooking.surname &&
+    newBooking.email &&
+    newBooking.roomId &&
+    newBooking.checkInDate &&
+    newBooking.checkOutDate
+  ) {
+    bookings.push(newBooking);
+    resp.send("New Booking has been added >>");
+  } else {
+    resp.status(400).send("Missin Information >> ");
+  }
+});
+//Read one booking, specified by an ID
+app.get("/booking/:id", function (request, response) {
+  const bookingID = +request.params.id;
+  const fliterBooking = bookings.find((booking) => booking.id === bookingID);
+  if (!fliterBooking) {
```

Comment:
I don't think you should use this syntax `!fliterBooking` in your if statement to validate because it might work when the find function returns a value. 
The only time you should use this syntax is when you know your function returns `true` or `false` but the `find` method does not return `true` or `false` but it returns a value or undefined so you should use undefined to compare in your if statement I believe.


You can read the "Return Value` on this section about the value that the find function retuns 
https://www.w3schools.com/jsref/jsref_find.asp#:~:text=The%20find()%20method%20returns,if%20no%20elements%20are%20found.